src/libs/graphs.js

```diff
@@ -20,107 +26,136 @@ function placeholders(vm, startVal) {
-            // } else {
-            //   console.log('WARNING: legenddiv not found: ' + legendlabel)
-            // }
-          }
-
-          // Draw figures
-          try {
-            mpld3.draw_figure(figlabel, graphdata[index], function (fig, element) {
-              fig.setXTicks(6, function (d) {
-                return d3.format('.0f')(d);
-              });
-              // fig.setYTicks(null, function (d) { // Looks too weird with 500m for 0.5
-              //   return d3.format('.2s')(d);
-              // });
-            }, true);
-          } catch (error) {
-            console.log('Could not plot graph: ' + error.message)
-          }
-
-          // Draw legends
-          // if (index>=1 && index<n_plots) {
-          //   try {
-          //     mpld3.draw_figure(legendlabel, legenddata[index], function (fig, element) {
-          //     });
-          //   } catch (error) {
-          //     console.log(error)
-          //   }
-          //
-          // }
-          vm.showGraphDivs[index] = true;
-        }
-        status.succeed(vm, 'Graphs created') // CK: This should be a promise, otherwise this appears before the graphs do
-      })
+
+  let waitingtime = 0.5
+  var graphdata = data.graphs
+  // var legenddata = data.legends
+
+  // Start indicating progress.
+  status.start(vm)
+
+  vm.hasGraphs = true
+  await utils.sleep(waitingtime * 1000);
+
+  let n_plots = graphdata.length
+  // let n_legends = legenddata.length
+
+  console.log('Rendering ' + n_plots + ' graphs')
+
+  // if (n_plots !== n_legends) {
+  //   console.log('WARNING: different numbers of plots and legends: ' + n_plots + ' vs. ' + n_legends)
+  // }
+
+  for (var index = 0; index <= n_plots; index++) {
+    console.log('Rendering plot ' + index);
+    var figlabel = 'fig' + index;
+
+    var figdiv  = document.getElementById(figlabel); 
+    if (!figdiv) {
+      console.log('WARNING: figdiv not found: ' + figlabel)
+    }
+
+    // Show figure containers
+    if (index>=1 && index < n_plots) {
+      var figcontainerlabel = 'figcontainer' + index
+
+      // CK: Not sure if this is necessary? To ensure the div is clear first
+      var figcontainerdiv = document.getElementById(figcontainerlabel); 
+      if (figcontainerdiv) {
+        figcontainerdiv.style.display = 'flex'
+      } else {
+        console.log('WARNING: figcontainerdiv not found: ' + figcontainerlabel)
+      }
+
+      // var legendlabel = 'legend' + index
+      // var legenddiv  = document.getElementById(legendlabel);
+      // if (legenddiv) {
+      //   while (legenddiv.firstChild) {
+      //     legenddiv.removeChild(legenddiv.firstChild);
+      //   }
+      // } else {
+      //   console.log('WARNING: legenddiv not found: ' + legendlabel)
+      // }
+    }
+
+    // Draw figures
+    try {
+      mpld3.draw_figure(figlabel, graphdata[index], function (fig, element) {
+        fig.setXTicks(6, function (d) {
+          return d3.format('.0f')(d);
```

Comment:
I know this isn't new in this PR, but can you remember why we had it? It breaks graphs with non-numeric tick marks, and in any case should be in mpld3, not sciris...

Reply:
I can't remember! I don't even know if it was me who wrote that.

Does that mean that you'd like to move that to `mpld3`?